PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/MavenProjectInspection.java

```diff
@@ -0,0 +1,97 @@
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.xml.XmlFile;
+import com.intellij.psi.xml.XmlTag;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+
+public class MavenProjectInspection extends XmlSuppressableInspectionTool {
+    @Nls
+    @NotNull
+    @Override
+    public String getDisplayName() {
+        return "Check Maven Project";
+    }
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new PsiElementVisitor() {
+
+            @Override
+            public void visitFile(@NotNull PsiFile file) {
+                if (!file.getName().equals("pom.xml")) {
+                    return;
+                }
+                MavenProjectsManager mavenProjectsManager = MavenProjectsManager.getInstance(file.getProject());
+                if (!mavenProjectsManager.isMavenizedProject()) {
+                    return;
+                }
+                FileViewProvider viewProvider = file.getViewProvider();
+                XmlFile xmlFile = (XmlFile) viewProvider.getPsi(StdLanguages.XML);
+
+                XmlTag rootTag = xmlFile.getRootTag();
+                if (rootTag != null && "project".equals(rootTag.getName())) {
+                    checkDependencyManagement(rootTag, holder);
+                    checkDependencies(rootTag, holder);
+                }
+            }
+        };
+    }
+
+    private static void checkDependencyManagement(XmlTag rootTag, @NotNull ProblemsHolder holder) {
+        XmlTag dependencyManagement = rootTag.findFirstSubTag("dependencyManagement");
+        if (dependencyManagement != null) {
+            XmlTag dependenciesTag = dependencyManagement.findFirstSubTag("dependencies");
+            if (dependenciesTag != null) {
+                XmlTag[] dependencyTags = dependenciesTag.findSubTags("dependency");
+                for (XmlTag dependencyTag : dependencyTags) {
+                    String groupId = dependencyTag.findFirstSubTag("groupId").getValue().getText();
+                    String artifactId = dependencyTag.findFirstSubTag("artifactId").getValue().getText();
+                    String versionId = null;
+
+                    if (dependencyTag.findFirstSubTag("version") != null) {
+                        versionId = dependencyTag.findFirstSubTag("version").getValue().getText();
+                    }
+
+                    if ("com.azure".equals(groupId) && artifactId.equals("azure-sdk-bom")) {
+                        if (!versionId.equals("1.2.24")) {
+                            holder.registerProblem(dependencyTag, "Newer version of azure-sdk-bom is available. Update to version 1.2.24");
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static void checkDependencies(XmlTag rootTag, @NotNull ProblemsHolder holder) {
```

Comment:
Given how IntelliJ inspections work, I wonder if it'd be better to split this into another inspection. These will be much nosier than just letting a user know there is a new version of the BOM, which may lead them to disable the inspection due to noise. But that would also disable the inspection for a newer BOM version.